source/features/remove-label-faster.tsx

```diff
@@ -0,0 +1,83 @@
+import './remove-label-faster.css';
+import React from 'dom-chef';
+import XIcon from 'octicon/x.svg';
+import select from 'select-dom';
+import oneTime from 'onetime';
+import delegate from 'delegate-it';
+import {observe} from 'selector-observer';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import * as api from '../github-helpers/api';
+
+import {getRepoURL, getConversationNumber} from '../github-helpers';
+
+const canNotEditLabels = oneTime((): boolean => !select.exists('.sidebar-labels .octicon-gear'));
+
+async function removeLabelButtonClickHandler(event: delegate.Event<MouseEvent, HTMLButtonElement>): Promise<void> {
+	event.preventDefault();
+
+	const removeLabelButton = event.delegateTarget;
+
+	removeLabelButton.disabled = true;
+	await api.v3(`repos/${getRepoURL()}/issues/${getConversationNumber()!}/labels/${removeLabelButton.dataset.name!}`, {
+		method: 'DELETE'
+	});
+
+	removeLabelButton.closest('a')!.remove();
+
+	const addLabelMenu = select('.sidebar-labels details-menu')!;
+	const addLabelMenuContent = addLabelMenu.querySelector('.hx_rsm-content');
+
+	if (addLabelMenuContent?.querySelector('include-fragment') === null) {
+		addLabelMenuContent.innerHTML = '';
+		addLabelMenuContent.append(<include-fragment src={addLabelMenu.getAttribute('src')!}/>);
+	}
```

Comment:
I saw the issue. Here are some minor improvements:

- adds explanatory comment
- only 1 query instead of 3, if `include-fragment` is still there
- avoids setting `src` (normally it's not there)
```suggestion
	// Force update of label selector if necessary
	if (!select.exists('.sidebar-labels include-fragment')) {
		const deferredContentWrapper = select('.sidebar-labels .hx_rsm-content');
		deferredContentWrapper.textContent = '';
		deferredContentWrapper.append(<include-fragment/>);
	}
```

Reply:
> * avoids setting `src` (normally it's not there)

![Peek 2020-08-13 22-06](https://user-images.githubusercontent.com/36174850/90176356-7a5bae00-ddb1-11ea-9078-5bd86b6f2dda.gif)

Reply:
Love the reverse engineering!

However the second point still applies, your code does 3 queries before it’s necessary. Let’s use my code and add `const menu = deferredContentWrapper.closest('[src]')!`